src/transformers/models/xglm/modeling_xglm.py

```diff
@@ -340,9 +339,15 @@ def forward(
                     f"Attention mask should be of size {(bsz, 1, tgt_len, src_len)}, but is {attention_mask.size()}"
                 )
             attn_weights = attn_weights.view(bsz, self.num_heads, tgt_len, src_len) + attention_mask
+            attn_weights = torch.max(attn_weights, torch.tensor(torch.finfo(attn_weights.dtype).min))
```

Comment:
@stas00 is this operation costly? Wondering how costly such a `max` operation is

Reply:
what are the contenders? at least `max`, `clamp` and `where`, but probably others as well?

```
$ cat clamp-where-max.py
import torch.utils.benchmark as benchmark
import torch

a = torch.tensor(512)
b = torch.tensor(torch.finfo(torch.float16).min)

t0 = benchmark.Timer(
    stmt='torch.clamp(a, b)',
    setup='',
    globals=dict(a=a, b=b),
)

t1 = benchmark.Timer(
    stmt='torch.max(a, b)',
    setup='',
    globals=dict(a=a, b=b),
)

t2 = benchmark.Timer(
    stmt='torch.where(a > b, a, b)',
    setup='',
    globals=dict(a=a, b=b),
)
print(t0.timeit(1000))
print(t1.timeit(1000))
print(t2.timeit(1000))

$ python clamp-where-max.py
<torch.utils.benchmark.utils.common.Measurement object at 0x7f73f6656040>
torch.clamp(a, b)
  2.41 us
  1 measurement, 1000 runs , 1 thread
<torch.utils.benchmark.utils.common.Measurement object at 0x7f73f6656d60>
torch.max(a, b)
  2.38 us
  1 measurement, 1000 runs , 1 thread
<torch.utils.benchmark.utils.common.Measurement object at 0x7f73f6656040>
torch.where(a > b, a, b)
  4.98 us
  1 measurement, 1000 runs , 1 thread
```

so `max` is slightly faster than `clamp`

but ensure to benchmark with the actual dimensions, though it shouldn't make much of a difference I think.